src/TriggerBinding/SqlTriggerBindingProvider.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Triggers;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql
+{
+    internal sealed class SqlTriggerBindingProvider : ITriggerBindingProvider
+    {
+        private readonly IConfiguration _configuration;
+        private readonly IHostIdProvider _hostIdProvider;
+        private readonly ILogger _logger;
+
+        public SqlTriggerBindingProvider(IConfiguration configuration, IHostIdProvider hostIdProvider, ILoggerFactory loggerFactory)
+        {
+            this._configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+            this._hostIdProvider = hostIdProvider ?? throw new ArgumentNullException(nameof(hostIdProvider));
+
+            _ = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
+            this._logger = loggerFactory.CreateLogger(LogCategories.CreateTriggerCategory("Sql"));
+        }
+
+        public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            ParameterInfo parameter = context.Parameter;
+            SqlTriggerAttribute attribute = parameter.GetCustomAttribute<SqlTriggerAttribute>(inherit: false);
+
+            if (attribute == null)
+            {
+                return Task.FromResult(default(ITriggerBinding));
```

Comment:
Shouldn't this throw? This doesn't seem like a valid scenario to return null here.

Reply:
This is how it is implemented in other extensions. Two examples: [Cosmos DB](https://github.com/Azure/azure-webjobs-sdk-extensions/blob/48fe14f0a373a3a279849eae0cf0b3756fa3d47f/src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs#L45-L48), [Service Bus](https://github.com/Azure/azure-functions-servicebus-extension/blob/dev/src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Triggers/ServiceBusTriggerAttributeBindingProvider.cs#L54-L57).

Reply:
Do you know why? I guess maybe it's because of the "Try" part of "TryCreateAsync" - is this method called for all trigger providers when a trigger binding attribute is detected or something and returning null is just saying "I don't know how to handle this type of trigger binding"? It would be great to document that here to make it clear why this is different from every other check we do where we throw.

Reply:
Here's the explanation: During startup, the WebJobs host calls [`CompositeTriggerBindingProvider.TryAsync`](https://github.com/Azure/azure-webjobs-sdk/blob/46514f23a5b5f3cc885d97a5c634f7f80e92a42e/src/Microsoft.Azure.WebJobs.Host/Triggers/CompositeTriggerBindingProvider.cs#L18-L30) which calls `TryAsync` on all the trigger binding providers registered for all the user function parameters. The first provider to return non-null binding is the one that is chosen whose `BindAsync` method will be called to construct the parameter value before every function invocation. Hence, the binding provider is supposed to return null if it does not find the trigger-attribute of its choice on the function parameter, so that next registered binding provider can get a chance.

Reply:
Yeah, sorta makes sense (although I'm curious why they don't have the binding attribute itself be used to determine which provider to use....).

Could you add a comment about that? Given that I've been unable to find any public docs about that behavior that seems like a useful thing to have to ensure that no one accidently removes it.

Reply:
Done in #288.